### PR TITLE
Customize S3's GetBucketLocation call to correctly parse the response

### PR DESCRIPTION
**Pull request by jdisanti**
This PR adds a custom Smithy trait to customize the XML deserialization codegen for S3's GetBucketLocation call to fix aws-sdk-rust#116.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rcoh** [comment]
Full review coming tomorrow–instead of adding it as an integration test, add it to s3-tests.smithy. that will make it easier to upstream

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/XmlBindingTraitParserGenerator.kt [hunk 1]
```diff
@@ -191,8 +192,12 @@ class XmlBindingTraitParserGenerator(
                     *codegenScope
                 )
                 val context = OperationWrapperContext(operationShape, shapeName, xmlError)
-                writeOperationWrapper(context) { tagName ->
-                    parseStructureInner(members, builder = "builder", Ctx(tag = tagName, accum = null))
+                if (outputShape.hasTrait<S3UnwrappedXmlOutputTrait>()) {
+                    unwrappedResponseParser("builder", "decoder", "start_el", outputShape.members())
```
Comment (rcoh) [inline codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/XmlBindingTraitParserGenerator.kt:196]: is there a reason you didn't just call `parseStructureInner` here? You could still validate the one member constraint here if you wanted

**Review by rcoh — APPROVED** [review]
LGTM (although see one question inline)! stoked that this came out so concise.

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/XmlBindingTraitParserGenerator.kt (hunk 1 above)
Comment (jdisanti) [reply codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/XmlBindingTraitParserGenerator.kt:196]: Yes, `parseStructInner` calls `parseLoop` which advances to the next tag (which won't exist in this case).

**Pull request merged**